### PR TITLE
Auto run d1 migrations

### PR DESCRIPTION
**Pull request by carlos-alberto**

### CODE REVIEW

File: src/test.ts [hunk 1]
```diff
@@ -23,38 +25,57 @@ export const setupTests = async <Paths extends {}>(bindings: Env) => {
...
 
+  // Hack until miniflare supports auto running D1 migrations.
+  // Note: Each migration file can contain only a single statement.
+  await Promise.all(
+    (config.d1_databases ?? []).map(
+      async ({ binding }: { binding: string }) => {
+        const d1 = await miniflare.getD1Database(binding);
+
+        const migrationsDir = "./migrations";
+        for (const migrationName of await nodefs.readdir(migrationsDir)) {
```
Comment (BarnabyShearer) [inline src/test.ts:68]: Wrangler applies them in a specific order
```
			.sort((a, b) => {
				const migrationNumberA = parseInt(a.name.split("_")[0]);
				const migrationNumberB = parseInt(b.name.split("_")[0]);
				if (migrationNumberA < migrationNumberB) {
					return -1;
				}
				if (migrationNumberA > migrationNumberB) {
					return 1;
				}

				// numbers must be equal
				return 0;
			});
```

File: src/test.ts [hunk 2]
```diff
@@ -23,38 +25,57 @@ export const setupTests = async <Paths extends {}>(bindings: Env) => {
...
   const url = await miniflare.ready;
 
+  // Hack until miniflare supports auto running D1 migrations.
+  // Note: Each migration file can contain only a single statement.
+  await Promise.all(
+    (config.d1_databases ?? []).map(
+      async ({ binding }: { binding: string }) => {
+        const d1 = await miniflare.getD1Database(binding);
+
+        const migrationsDir = "./migrations";
```
Comment (BarnabyShearer) [inline src/test.ts:67]: This is configurable; but :shrug:

File: src/test.ts [hunk 3]
```diff
@@ -23,38 +25,57 @@ export const setupTests = async <Paths extends {}>(bindings: Env) => {
...
+        for (const migrationName of await nodefs.readdir(migrationsDir)) {
+          const migrationPath = path.join(migrationsDir, migrationName);
+          let migration = await nodefs.readFile(migrationPath, "utf8");
+          // Remove migration comment
+          migration = migration.split("\n").slice(1).join(" ");
```
Comment (BarnabyShearer) [inline src/test.ts:72]: Feels like we are going to trip over more different syntaxes here, we may need to use wrangler to give us more parity; we can try and see.

**Review by BarnabyShearer — CHANGES_REQUESTED** [review]

File: src/test.ts (hunk 2 above)
Comment (carlos-alberto) [reply src/test.ts:67]: > flexible is basically an anti-feature in two-stroke

**Review by BarnabyShearer — APPROVED** [review]

**Pull request merged**